### PR TITLE
NISP-2297: Added connector methods and tests

### PR DESCRIPTION
**Pull request by prashantchoudhari**
Hi @dspork 
Please check the liability container specifically
and of course other things :)

### CODE REVIEW

File: app/uk/gov/hmrc/nationalinsurancerecord/connectors/NpsConnector.scala [hunk 1]
```diff
@@ -16,12 +16,90 @@
...
+  def http: HttpGet
+  def url(path: String): String = s"$serviceUrl$path"
+  def requestHeaderCarrier(implicit hc: HeaderCarrier): HeaderCarrier = hc.withExtraHeaders(serviceOriginatorIdKey -> serviceOriginatorId)
+  private def ninoWithoutSuffix(nino: Nino): String = nino.value.substring(0, NIRecordConstants.ninoLengthWithoutSuffix)
+
+  def getLiabilities(nino: Nino)(implicit hc: HeaderCarrier): Future[NpsLiabilities] = {
+    val urlToRead = url(s"/nps-rest-service/services/nps/pensions/${ninoWithoutSuffix(nino)}/liabilities")
+    connectToNps[NpsLiabilities](urlToRead, APITypes.Liabilities, requestHeaderCarrier) map {
+      response =>
+        response
```
Comment (dspork) [inline app/uk/gov/hmrc/nationalinsurancerecord/connectors/NpsConnector.scala]: This map does not do anything.

File: app/uk/gov/hmrc/nationalinsurancerecord/connectors/NpsConnector.scala [hunk 2]
```diff
@@ -16,12 +16,90 @@
...
+  val serviceOriginatorId: String
+
+  class JsonValidationException(message: String) extends Exception(message)
+
+  def http: HttpGet
+  def url(path: String): String = s"$serviceUrl$path"
+  def requestHeaderCarrier(implicit hc: HeaderCarrier): HeaderCarrier = hc.withExtraHeaders(serviceOriginatorIdKey -> serviceOriginatorId)
+  private def ninoWithoutSuffix(nino: Nino): String = nino.value.substring(0, NIRecordConstants.ninoLengthWithoutSuffix)
+
+  def getLiabilities(nino: Nino)(implicit hc: HeaderCarrier): Future[NpsLiabilities] = {
```
Comment (dspork) [inline app/uk/gov/hmrc/nationalinsurancerecord/connectors/NpsConnector.scala]: Personally I would change this method to return the List[NpsLiabilities] and map the response in this method.  But that's not required.

File: app/uk/gov/hmrc/nationalinsurancerecord/connectors/NpsConnector.scala [hunk 3]
```diff
@@ -16,12 +16,90 @@
...
+      response =>
+        response
+    }
+  }
+
+  def getSummary(nino: Nino)(implicit hc: HeaderCarrier): Future[NpsSummary] = {
+    val urlToRead = url(s"/nps-rest-service/services/nps/pensions/${ninoWithoutSuffix(nino)}/sp_summary")
+    connectToNps[NpsSummary](urlToRead, APITypes.Summary, requestHeaderCarrier) map {
+      response =>
+        response
```
Comment (dspork) [inline app/uk/gov/hmrc/nationalinsurancerecord/connectors/NpsConnector.scala]: This map does not do anything.

File: app/uk/gov/hmrc/nationalinsurancerecord/connectors/NpsConnector.scala [hunk 4]
```diff
@@ -16,12 +16,90 @@
...
+    }
+  }
+
+
+  def getNationalInsuranceRecord(nino: Nino)(implicit hc: HeaderCarrier): Future[NpsNIRecord] = {
+    val urlToRead = url(s"/nps-rest-service/services/nps/pensions/${ninoWithoutSuffix(nino)}/ni_record")
+    connectToNps[NpsNIRecord](urlToRead, APITypes.NIRecord, requestHeaderCarrier) map {
+      response =>
+        response
```
Comment (dspork) [inline app/uk/gov/hmrc/nationalinsurancerecord/connectors/NpsConnector.scala]: This map does not do anything.

**Comment by prashantchoudhari** [comment]
@dspork - removed maps by @PaulCDurham

**Pull request merged**